packages/service-library/tests/long_running_tasks/test_long_running_tasks_task.py

```diff
@@ -0,0 +1,290 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+
+import asyncio
+from datetime import datetime
+from typing import AsyncIterator, Final
+
+import pytest
+from servicelib.long_running_tasks._errors import (
+    TaskAlreadyRunningError,
+    TaskCancelledError,
+    TaskNotCompletedError,
+    TaskNotFoundError,
+)
+from servicelib.long_running_tasks._models import TaskProgress, TaskResult, TaskStatus
+from servicelib.long_running_tasks._task import TasksManager, start_task
+from tenacity._asyncio import AsyncRetrying
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_fixed
+
+# UTILS
+
+
+async def a_background_task(
+    task_progress: TaskProgress,
+    raise_when_finished: bool,
+    total_sleep: int,
+) -> int:
+    """sleeps and raises an error or returns 42"""
+    for i in range(total_sleep):
+        await asyncio.sleep(1)
+        task_progress.publish(percent=float((i + 1) / total_sleep))
+    if raise_when_finished:
+        raise RuntimeError("raised this error as instructed")
+
+    return 42
+
+
+async def fast_background_task(task_progress: TaskProgress) -> int:
+    """this task does nothing and returns a constant"""
+    return 42
+
+
+async def failing_background_task(task_progress: TaskProgress):
+    """this task does nothing and returns a constant"""
+    raise RuntimeError("failing asap")
+
+
+TEST_CHECK_STALE_INTERVAL_S: Final[float] = 1
+
+
+@pytest.fixture
+async def tasks_manager() -> AsyncIterator[TasksManager]:
+    tasks_manager = TasksManager(
+        stale_task_check_interval_s=TEST_CHECK_STALE_INTERVAL_S,
+        stale_task_detect_timeout_s=TEST_CHECK_STALE_INTERVAL_S,
+    )
+    yield tasks_manager
+    await tasks_manager.close()
+
+
+async def test_unchecked_task_is_auto_removed(tasks_manager: TasksManager):
+    task_id = start_task(
+        tasks_manager,
+        a_background_task,
+        raise_when_finished=False,
+        total_sleep=10 * TEST_CHECK_STALE_INTERVAL_S,
+    )
+    await asyncio.sleep(2 * TEST_CHECK_STALE_INTERVAL_S + 1)
+    with pytest.raises(TaskNotFoundError):
+        tasks_manager.get_task_status(task_id)
+    with pytest.raises(TaskNotFoundError):
+        tasks_manager.get_task_result(task_id)
+    with pytest.raises(TaskNotFoundError):
+        tasks_manager.get_task_result_old(task_id)
```

Comment:
What about the delete endpoint? Should that one not raise the same?

Reply:
yes and there is a test specifically for it at line 286